OSXvnc-server/main.c

```diff
@@ -563,13 +570,35 @@ char *rfbGetFramebuffer(void) {
     if (floor(NSAppKitVersionNumber) > floor(NSAppKitVersionNumber10_6)) {
         if (!frameBufferData) {
             CGDirectDisplayID mainDisplayID = CGMainDisplayID();
-            CGImageRef imageRef = CGDisplayCreateImage(mainDisplayID);
+            CGImageRef imageRef;
+            //Check to see if Retina Display
+            if (floor(NSAppKitVersionNumber) > floor(NSAppKitVersionNumber10_6) & displayScale > 1) {
```

Comment:
You are correct to a point.  You can simulate scaled displays in xcode and some screen resolution apps under snow leopard.  But I am fine if it is removed